qutebrowser/mainwindow/prompt.py

```diff
@@ -697,6 +706,23 @@ def _insert_path(self, index, *, clicked=True):
             # On Windows, when we have C:\foo and tab over .., we get C:\
             path = path.rstrip(os.sep)
 
+        c_sep = 0
+        index_end_home = 0
+        for i, character in enumerate(path):
+            if character in self._separators:
+                c_sep += 1
+
+            index_end_home = i
+
+            if c_sep == 3:
+                break
+
+        if self._expands_user:
+            path = (path[:index_end_home].replace(os.path.expanduser('~'), '~') +
+                    path[index_end_home:])
+        else:
+            self._expands_user = False
```

Comment:
If I have a path `/home/user/test/home/user` what this section does is replace `/home/user` only from the beginning with `~` so it will give you `~/user/test/home/user` rather than `~/test/~` if I didn't get the first couple sections of the path.

Reply:
But `~` doesn't necessarily expand to `/home/user`. qutebrowser is cross-platform, and on e.g. Windows, this will expand to `C:\Users\username`; on macOS it'll expand to `/Users/username`, etc. etc.

Are you perhaps just looking for the `count` option for Python's [str.replace](https://docs.python.org/3.8/library/stdtypes.html#str.replace)?

Finally, this will need a test case as well.